openghg/dataobjects/_searchresults.py

```diff
@@ -200,79 +217,49 @@ def retrieve(
         species = clean_string(species)
         inlet = clean_string(inlet)
 
-        # If inlet is not specified, check if this is unambiguous
-        # If so, set inlet to be the only value and continue.
-        if inlet is None:
-            try:
-                potential_inlets = self.results[site][species].keys()
-            except KeyError:
-                pass
-            else:
-                if len(potential_inlets) == 1:
-                    inlet = list(potential_inlets)[0]
+        results = []
+        if not self.ranked_data:
+            for _site, site_data in self.results.items():
+                if site is not None and _site != site:
+                    continue
+                for _species, species_data in site_data.items():
+                    if species is not None and _species != species:
+                        continue
+                    for _inlet in species_data:
+                        if inlet is not None and _inlet != inlet:
+                            continue
+
+                        obsdata = self._create_obsdata(site=_site, species=_species, inlet=_inlet)
+
+                        if obsdata is not None:
+                            results.append(obsdata)
+        else:
+            if inlet is not None:
```

Comment:
I may be mis-interpreting this but if someone passes an inlet and the data is ranked can we just bypass the ranking to return that specific inlet? I'm just thinking you may want to retrieve data from a particular inlet regardless of ranking e.g. for comparison between inlets.